src/js/common/components/NavButtons.jsx

```diff
@@ -21,26 +25,42 @@ export default class NavButtons extends React.Component {
       this.props.onSubmit();
     }
   }
-  render() {
-    const { submission, path, pages } = this.props;
-    const currentIndex = pages.indexOf(path);
-    const nextPath = currentIndex + 1 < pages.length ? pages[currentIndex + 1] : null;
-    const previousPath = currentIndex - 1 >= 0 ? pages[currentIndex - 1] : null;
+  goBack() {
+    this.props.onNavigate(this.findNeighbor(-1));
+  }
+  goForward() {
+    this.handleContinue(this.findNeighbor(1));
+  }
+  findNeighbor(increment) {
+    const { pages, path, data } = this.props;
+    const currentIndex = pages.map(page => page.name).indexOf(path);
 
-    const goBack = () => this.props.onNavigate(previousPath);
-    const goForward = () => this.handleContinue(nextPath);
+    for (let i = currentIndex + increment; i >= 0 && i < pages.length; i += increment) {
```

Comment:
I was thinking "there must be a lodash function that does this..."  Sure enough!

I do think the second approach is better.  I might wrap that up into a function that gets called on button click.

(My goal was to avoid running it on render because `_.matches` has the possibility of being slow if it's matching a lot of data.)